### PR TITLE
identify 3

### PR DESCRIPTION
**Pull request by chrisnojima**
JS:
- [x] profile screen
- [x] team popups
- [x] search bar
- [x] you are in team 
- [x] stellar assertions
- [x] ~you requested access plumbing in popup? (not doing this)~
- [x] folders list
- [x] ... menu
- [x] edit bio desktop
- [x] edit bio mobile
- [x] edit avatar desktop
- [x] edit avatar mobile
- [x] edit your teams
- [x] edit teams
- [x] select teams popup

Core: (cc: @maxtaco)
- [ ] siteIcons need to work. (i'm mapping from key to the font icons but this needs to change soon)

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/tracker2/remote-serializer.desktop.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  users: {
+    infoMap: I.Map(),
+  },
+  waiting: {
+    counts: I.Map(),
+  },
+}
+
+export const deserialize = (state: any = initialState, props: any) => {
+  console.log('aaaa deserialize', state, props)
```
Comment (maxtaco) [inline shared/tracker2/remote-serializer.desktop.js]: get rid of this?

**Comment by buoyad** [comment]
Worth noting that the feature flag needs to be turned on for reviewing https://github.com/keybase/client/blob/457bd67dd844b1e578fb036181b4c693d07ee21d/shared/util/feature-flags.desktop.js#L17

**Comment by chrisnojima** [comment]
yeah sorry, also you should test that it works w/ the flag off also

File: shared/tracker2/index.desktop.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
+// @flow
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/tracker2/index.desktop.js]: I also got this black bar once as the tracker opened
```
Error: update user card w/ no username? 8e3fbcdb0b94b 
at updateUserCard (webpack:///./actions/tracker2.js?:90:11) 
at runCallEffect (webpack:///./node_modules/redux-saga/es/internal/proc.js?:523:19) 
at runEffect (webpack:///./node_modules/redux-saga/es/internal/proc.js?:445:594) 
at next (webpack:///./node_modules/redux-saga/es/internal/proc.js?:325:9) 
at currCb (webpack:///./node_modules/redux-saga/es/internal/proc.js?:398:7) 
at runSelectEffect (webpack:///./node_modules/redux-saga/es/internal/proc.js?:709:7) 
at runEffect (webpack:///./node_modules/redux-saga/es/internal/proc.js?:445:1113) 
at next (webpack:///./node_modules/redux-saga/es/internal/proc.js?:325:9) 
at proc (webpack:///./node_modules/redux-saga/es/internal/proc.js?:280:3) 
at runForkEffect (webpack:///./node_modules/redux-saga/es/internal/proc.js?:564:19)
```

[the remark this answers is not in the document]
Comment (buoyad) [reply shared/tracker2/index.desktop.js]: Also, this closes if I open up the same profile in the app, is that intended?

**Comment by chrisnojima** [comment]
@buoyad updated. i made a theoretical fix for the slow close. i can't get it to be slow

**Comment by chrisnojima** [comment]
oh also make sure you do a build of the daemon as this has some small fixes in it not on master

**Comment by chrisnojima** [comment]
@buoyad updated

**Comment by buoyad** [comment]
Trackers with the identify3 flag off show up like this:
image

File: shared/actions/tracker2.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  yield* Saga.chainAction<EngineGen.Keybase1Identify3UiIdentify3UpdateUserCardPayload>(
+    EngineGen.keybase1Identify3UiIdentify3UpdateUserCard,
+    updateUserCard
+  )
+  yield* Saga.chainAction<Tracker2Gen.ChangeFollowPayload>(Tracker2Gen.changeFollow, changeFollow)
+  yield* Saga.chainAction<Tracker2Gen.IgnorePayload>(Tracker2Gen.ignore, ignore)
+  yield* Saga.chainGenerator<Tracker2Gen.LoadPayload>(Tracker2Gen.load, load)
+  yield* Saga.chainAction<Tracker2Gen.LoadPayload>(Tracker2Gen.load, loadFollow)
+
+  // TEMP until actions/tracker is deprecated
```
Comment (buoyad) [inline shared/actions/tracker2.js]: should this be before the `if (!flags.identify3)`?

File: shared/profile/user/teams/index.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+)
+
+const Teams = (p: Props) =>
+  p.onEdit || p.teamShowcase.length > 0 ? (
+    <Kb.Box2 direction="vertical" gap="tiny" fullWidth={true} style={styles.showcases}>
+      <Kb.Box2 direction="horizontal" gap="tiny" fullWidth={true}>
+        <Kb.Text type="BodySmallSemibold">Teams</Kb.Text>
+        {!!p.onEdit && <Kb.Icon type="iconfont-edit" onClick={p.onEdit} />}
+      </Kb.Box2>
+      {!!p.onEdit && <ShowcaseTeamsOffer onEdit={p.onEdit} />}
```
Comment (buoyad) [inline shared/profile/user/teams/index.js]: `&& !p.teamShowcase.length`

File: shared/common-adapters/usernames/index.js [hunk 5]
```diff
@@ -19,6 +19,7 @@ export type BaseUsernamesProps = {|
   backgroundMode?: Background,
```
Comment (buoyad) [inline shared/common-adapters/usernames/index.js]: I show up as orange with the flag off, intended?

File: shared/actions/tracker2.js (hunk 3 above)
Comment (chrisnojima) [reply shared/actions/tracker2.js]: don't think so, tracker1 handles it itself

File: shared/profile/user/teams/index.js (hunk 4 above)
Comment (chrisnojima) [reply shared/profile/user/teams/index.js]: we show this even if you have showcased teams on master right

Comment (buoyad) [reply shared/profile/user/teams/index.js]: @chrisnojima no

File: shared/common-adapters/usernames/index.js (hunk 5 above)
Comment (chrisnojima) [reply shared/common-adapters/usernames/index.js]: fixed, actually colorYou was being set to orange always in name with icon

File: shared/profile/user/teams/index.js (hunk 4 above)
Comment (chrisnojima) [reply shared/profile/user/teams/index.js]: tnx done

File: shared/tracker2/assertion/container.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import Assertion from '.'
+import openUrl from '../../util/open-url'
+
+type OwnProps = {|
+  username: string,
+  assertionKey: string,
+|}
+
+const mapStateToProps = (state, ownProps) => {
+  const d = state.tracker2.usernameToDetails.get(ownProps.username, Constants.noDetails)
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/tracker2/assertion/container.js]: done

**Comment by chrisnojima** [comment]
@buoyad tnx. put in the feature flag for admins and doing shots now

**Comment by chrisnojima** [comment]
- [ ] jenkins changes
- [ ] temp stuff in stories to debug

**Comment by chrisnojima** [comment]
@buoyad i had to fix up some stuff 
- [x] sorting was using the wrong field https://github.com/keybase/client/pull/15231/commits/c4e79bc46187012f2c5364f432346b87ab69c0ae#diff-e39e893a60f50cb6e682d69494418db8R144
- [x] have to make a copy before sort https://github.com/keybase/client/pull/15231/commits/64c722748dc3b93b8da3f2e79c4cb9e0c2a3b246#diff-b7ceeafe0f9196817f0a95283b28a0a3L80

**Pull request merged**